lib/controllers/embed.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+var winston = module.parent.require('winston');
+
+var nbbAuthController = require.main.require('./src/controllers/authentication');
+var jwt = require('jsonwebtoken');
+var utils = require('../../utils');
+var meta = require.main.require('./src/meta');
+
+
+var embedControllers = module.exports;
+
+embedControllers.embedView = function (req, res, next) {
+	meta.settings.get('openedx-discussion', function (err, settings) {
+		if (err) {
+			return next({
+				code: 'error',
+				plugin: 'openedx-discussion',
+				message: 'Settings could not be loaded',
+			});
+		}
+		if (!settings.hasOwnProperty('secret') || !settings.secret.length ||
+			!settings.hasOwnProperty('jwtCookieName') || !settings.jwtCookieName.length) {
+			const message = 'openedx-discussion is not configured correctly.';
```

Comment:
Add what is wrong with the configuration in this message.

Reply:
we get the value for `secret` and the `cookieName` which contains the encoded token from user through admin panel of plugin. Therefore if these settings are not set first, cookie cannot be decoded and verified.

Reply:
I know that. But please add which setting or parameter is misconfigured in the log message. So someone who is going through the logs can check what is wrong with his/her settings.